app/application/route.js

```diff
@@ -75,19 +75,33 @@ export default Route.extend({
     // options will be the customizations added to the cart item, such as
     // quantity, color, size, etc.
     addToCart(product, options = {}) {
-      this.store.createRecord('cart-item', {
-        product,
-        quantity: parseInt(options.quantity, 10)
-      });
+      const cartItems = this.store.peekAll('cart-item');
+      const productQuantity = options.quantity || 1;
+      let cartItem;
+
+      if (cartItems.isAny('product.id', product.get('id'))) {
```

Comment:
could you avoid manually finding the object with `findBy`?

```js
const cartItem = cartItems.findBy('id', product.get('id'));
cartItem.incrementProperty('quantity', productQuantity);
```

Reply:
done ✅